lib/honeycomb/propagation.rb

```diff
@@ -47,17 +47,35 @@ def parse_v1(payload)
 
       [trace_id, parent_span_id, trace_fields, dataset]
     end
+
+    private
+
+    def parse_span_id(string)
+      INVALID_SPAN_ID = ("0" * 16).b
+      raise "invalid span id" if string == INVALID_SPAN_ID
+
+      value.downcase!
+      Array(value).pack('H*')
+
+    def parse_trace_id(string)
+      INVALID_TRACE_ID = ("0" * 32).b
+      raise "invalid trace id" if string == INVALID_TRACE_ID
+
+      value.downcase!
```

Comment:
where is value coming from here?

Reply:
that was from me initially doing this work inline above, and then opting to pull out the logic and pushing it up too fast. Sorry! Fixed